src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -0,0 +1,478 @@
+              #                               stderr=workunit.output('stderr'))
+
+              workunit.set_outcome(WorkUnit.FAILURE if return_code else WorkUnit.SUCCESS)
+
+              if return_code:
+                raise TaskError('The coursier process exited non-zero: {0}'.format(return_code))
+
+              with open(output_fn) as f:
+                result = json.loads(f.read())
+
+          except subprocess.CalledProcessError as e:
+            raise CoursierError(e)
+
+          else:
+            flattened_resolution = self._flatten_resolution_by_root(result)
+            files_by_coord = self._map_coord_to_resolved_jars(result, coursier_cache_path, pants_jar_path_base)
+
+            org_name_to_org_name_rev = {}
+            for coord in files_by_coord.keys():
+              (org, name, _) = coord.split(':')
+              org_name_to_org_name_rev['{}:{}'.format(org, name)] = coord
+            for vt in invalidation_check.all_vts:
+              t = vt.target
+            # for t in targets:
+              if isinstance(t, JarLibrary):
+
+                def get_transitive_resolved_jars(my_simple_coord, resolved_jars):
+                  transitive_jar_path_for_coord = []
+                  if my_simple_coord in flattened_resolution:
+                    for c in [my_simple_coord] + flattened_resolution[my_simple_coord]:
+                      transitive_jar_path_for_coord.extend(resolved_jars[c])
+
+                  return transitive_jar_path_for_coord
+
+                for jar in t.jar_dependencies:
+                  simple_coord_candidate = jar.coordinate.simple_coord
+                  final_simple_coord = None
+                  if simple_coord_candidate in files_by_coord:
+                    final_simple_coord = simple_coord_candidate
+                  elif simple_coord_candidate in result['conflict_resolution']:
+                    final_simple_coord = result['conflict_resolution'][simple_coord_candidate]
+                  # If still not found, look for org:name match.
+                  else:
+                    org_name = '{}:{}'.format(jar.org, jar.name)
+                    if org_name in org_name_to_org_name_rev:
+                      final_simple_coord = org_name_to_org_name_rev[org_name]
+
+                  if final_simple_coord:
+                    transitive_resolved_jars = get_transitive_resolved_jars(final_simple_coord, files_by_coord)
+                    if transitive_resolved_jars:
+                      compile_classpath.add_jars_for_targets([t], 'default' or classifier, transitive_resolved_jars)
+
+        self._write_result_to_cache(compile_classpath, invalidation_check.all_vts, target_resolution_filename)
+
+  def _write_result_to_cache(self, compile_classpath, all_vts, target_resolution_filename):
+    # TODO(wisechengyi): currently the path contains abs path, so need to remove that before
+    # cache can be shared across machines.
```

Comment:
If that's the case, then you should not write to the cache at all, as if a task is writing to the cache it will go in the cache unless users make settings changes... and that should not be required.

Reply:
renamed the methods to write to/load from results dir. no cache is involved.